qiskit/dagcircuit/dagnode.py

```diff
@@ -102,27 +102,41 @@ def __str__(self):
         return str(id(self))
 
     @staticmethod
-    def semantic_eq(node1, node2):
+    def semantic_eq(node1, node2, bit_indices1, bit_indices2):
```

Comment:
I'm a bit concerned about this change from a backwards compat and interop PoV. Like someone could have been using semantic eq to compare dag nodes outside of `DAGCircuit.__eq__` before right and this will now break them? Maybe making them optional and just having a fallback to `node1_qargs = node1._qargs` if it's not specified.

Reply:
That makes sense. I'll add that (with a deprecation warning).

Reply:
Added in  0dc3d7af82 .